frontend/src/app/components/MainInstallButton.js

```diff
@@ -55,15 +56,28 @@ export default class MainInstallButton extends React.Component {
       installButton = <span className="default-btn-msg" data-l10n-id="landingInstallButton">
       </span>;
     }
+    const makeInstallButton = (extraClass = '') => {
+      return <button onClick={e => this.install(e)}
+        className={classnames(`button extra-large primary install ${extraClass}`, { 'state-change': isInstalling })}>
+        {hasAddon && <span className="progress-btn-msg" data-l10n-id="landingInstalledButton">Installed</span>}
+        {!hasAddon && !isInstalling && installButton}
+        {!hasAddon && isInstalling &&
+          <span className="progress-btn-msg" data-l10n-id="landingInstallingButton">Installing...</span>}
+        <div className="state-change-inner"></div>
+      </button>;
+    };
+
     return (
-        <button onClick={e => this.install(e)}
-                className={classnames('button extra-large primary install', { 'state-change': isInstalling })}>
-          {hasAddon && <span className="progress-btn-msg" data-l10n-id="landingInstalledButton">Installed</span>}
-          {!hasAddon && !isInstalling && installButton}
-          {!hasAddon && isInstalling &&
-            <span className="progress-btn-msg" data-l10n-id="landingInstallingButton">Installing...</span>}
-          <div className="state-change-inner"></div>
-        </button>
+      <div>
```

Comment:
Does this need the bonus wrapper `<div>` or can we promote `<VariantTests>` and avoid a nested DOM element?